flake.nix

```diff
@@ -6,7 +6,7 @@
 
     # Backend inputs
     shared-kernel = {
-      url = "github:nammayatri/shared-kernel";
+      url = "github:nammayatri/shared-kernel/handling-kafka-push-for-critical-path";
```

Comment:
don't have to route to a specific branch you can run:

`nix flake lock --update-input shared-kernel --override-input shared-kernel github:nammayatri/shared-kernel/<COMMIT_HASH_YOU_WANT_TO_POINT_TO>`